interp/interp_test.go

```
@@ -1038,3 +1041,34 @@ func TestRunnerOpts(t *testing.T) {
 		})
 	}
 }
+
+func TestContext(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
+	defer cancel()
+
+	p, err := syntax.Parse(strings.NewReader("while true; do true; done;"), "", 0)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	buff := bytes.NewBuffer(nil)
+	r := Runner{
+		File:    p,
+		Stdout:  buff,
+		Context: ctx,
+	}
+	endChan := make(chan struct{})
+
+	go func() {
+		if err := r.Run(); err != nil {
+			t.Fatal(err)
```

I believe you can't run `t.Fatal` from a goroutine... Just ignore the error.